etna/models/nn/mlp.py

```diff
@@ -0,0 +1,217 @@
+        """
+        super().__init__()
+        self.input_size = input_size
+        self.hidden_size = hidden_size
+        self.lr = lr
+        self.loss = nn.MSELoss() if loss is None else loss
+        self.optimizer_params = {} if optimizer_params is None else optimizer_params
+        layers = [nn.Linear(in_features=input_size, out_features=hidden_size[0]), nn.ReLU()]
+        for i in range(1, len(hidden_size)):
+            layers.append(nn.Linear(in_features=hidden_size[i - 1], out_features=hidden_size[i]))
+            layers.append(nn.ReLU())
+        layers.append(nn.Linear(in_features=hidden_size[-1], out_features=1))
+        self.mlp = nn.Sequential(*layers)
+
+    def forward(self, batch):
+        """Forward pass.
+
+        Parameters
+        ----------
+        batch:
+            batch of data
+        Returns
+        -------
+        :
+            forecast
+        """
+        decoder_real = batch["decoder_real"].float()
+        return self.mlp(decoder_real)
+
+    def step(self, batch: MLPBatch, *args, **kwargs):  # type: ignore
+        """Step for loss computation for training or validation.
+
+        Parameters
+        ----------
+        batch:
+            batch of data
+        Returns
+        -------
+        :
+            loss, true_target, prediction_target
+        """
+        decoder_real = batch["decoder_real"].float()
+        decoder_target = batch["decoder_target"].float()
+
+        output = self.mlp(decoder_real)
+        loss = self.loss(output, decoder_target)
+        return loss, decoder_target, output
+
+    def make_samples(self, df: pd.DataFrame, encoder_length: int, decoder_length: int) -> Iterable[dict]:
+        """Make samples from segment DataFrame."""
+
+        def _make(df: pd.DataFrame, start_idx: int, decoder_length: int) -> Optional[dict]:
+            sample: Dict[str, Any] = {"decoder_real": list(), "decoder_target": list(), "segment": None}
+            total_length = len(df["target"])
+            total_sample_length = decoder_length
+
+            if total_sample_length + start_idx > total_length:
+                return None
+
+            sample["decoder_real"] = (
+                df.select_dtypes(include=[np.number])
+                .pipe(lambda x: x[[i for i in x.columns if i != "target"]])
+                .values[start_idx : start_idx + decoder_length]
+            )
+
+            target = df["target"].values[start_idx : start_idx + decoder_length].reshape(-1, 1)
+            sample["decoder_target"] = target
+            sample["segment"] = df["segment"].values[0]
+            return sample
+
+        start_idx = 0
+        while True:
+            batch = _make(
+                df=df,
+                start_idx=start_idx,
+                decoder_length=decoder_length,
+            )
+            if batch is None:
+                break
+            yield batch
+            start_idx += 1
```

Comment:
We should increment by decoder_lenght. For now every point sampled `decoder_lenght` times